lib/libc/Kconfig

```diff
@@ -29,6 +29,15 @@ config MINIMAL_LIBC
 	help
 	  Build with minimal C library.
 
+config PICOLIBC
+	bool "Picolibc library"
+	depends on !NATIVE_APPLICATION
+	select THREAD_LOCAL_STORAGE if ARCH_HAS_THREAD_LOCAL_STORAGE
```

Comment:
A small question trying to open a chat to see if this is right path.

Initially it appears wrong to me that the libc Kconfig should know such details of the Kernel config.

We have a `TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` here:
https://github.com/zephyrproject-rtos/zephyr/blob/20b3d180d1e998a84b6ae7dcb45301d3f98bceb3/kernel/Kconfig#L888

so maybe we should ensure that this value defaults to `y` when a toolchain with picolib is enabled.

Extra question, will using `PICOLIBC` break if `TREAD_LOCAL_STORAGE` is disabled ?
Reason i'm asking is because that will indicate whether a select or `default y if XYZ` is the best way to go.

Reply:
The only support picolibc has for thread local variables, including 'errno', is by using the underlying native TLS support. If the toolchain doesn't have TLS support, then all thread local variables in picolibc are simple globals.

As a result, by default, picolibc uses TLS whenever the toolchain has TLS support. What Zephyr needs to do is enable TLS whenever picolibc was built to use it.

We could check whether picolibc was built to use TLS by looking at the picolibc.h header file to see if PICOLIBC_TLS is defined, but I'm not sure how to do that with cmake. Right now, the checks I've added assume that picolibc was built in the default mode and uses TLS iff the toolchain has TLS support.

I'd appreciate whatever help you can offer to clean up the cmake bits here; I'm very unsure of how to use cmake in complicated projects like this.

Reply:
fwiw, this review series relates to changes proposed in #39563